### PR TITLE
Handle screen buffer scrolling correctly for dynamic in-line help rendering

### LINKED ISSUE #2950: Dynamic help rendering behaves incorrectly when buffer scrolling is required
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSReadLine)
- [X] Search the existing issues, especially the pinned issues.

### Exception report

```console
No exception, but the rendering is incorrect.
```

### Screenshot

As is shown in the GIF below, the cursor position is off when the dynamic help rendering needs buffer scroll-up.
This is because when it comes to adjust the cursor position, the code doesn't take into account the extra physical lines introduced by multi-line text.

Animation

### Environment data

```console
PS Version: 7.1.3
PS HostName: ConsoleHost (Windows Terminal)
PSReadLine Version: 2.2.0-beta4
PSReadLine EditMode: Windows
OS: 10.0.19041.1 (WinBuild.160101.0800)
BufferWidth: 133
BufferHeight: 30
```

### Steps to reproduce

1. Type a number of <kbd>Enter</kbd> to move the prompt to the last line of the screen buffer.
2. Copy `get-Command -Name` to the line
3. Press <kbd>Alt+h</kbd> to trigger the dynamic in-line help
4. Press any key to clear the content, and then press <kbd>Alt+h</kbd> again.

You will see the cursor position becomes incorrect.

### Expected behavior

Cursor position should be at the end of `get-Command -Name`

### Actual behavior

Cursor position becomes incorrect.

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Fix #2950

Major changes:
1. Update the variable name to make it more clear in meaning.
2. Adjust for buffer scrolling in case there are extra physical lines.

It works as expected after the fix:

Animation

## PR Checklist

- [x] PR has a meaningful title
    - Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [ ] Make sure you've added one or more new tests
      - **Cannot add tests because we cannot mimic buffer scrolling with the testing console, but I manually validated the fix**
- [x] Make sure you've tested these changes in terminals that PowerShell is commonly used in (i.e. conhost.exe, Windows Terminal, Visual Studio Code Integrated Terminal, etc.)
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] Documentation needed at [PowerShell-Docs](https://github.com/MicrosoftDocs/PowerShell-Docs)
        - [ ] Doc Issue filed: 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/PowerShell/PSReadLine/pull/2951)

### CODE REVIEW

**Comment by daxian-dbw** [comment]
@adityapatwardhan @andschwa, can one of you please do a sanity check for me? Thanks!

File: PSReadLine/DynamicHelp.cs [hunk 1]
```diff
@@ -261,16 +262,23 @@ public void DrawMultilineBlock()
...
                         {
-                            multilineItems--;
+                            extra--;
                         }
                     }
 
+                    if (extra > 0)
+                    {
+                        // Extra physical lines may cause buffer to scroll up.
+                        AdjustForPossibleScroll(extra);
```
Comment (andyleejordan) [inline PSReadLine/DynamicHelp.cs:278]: 👍

**Review by andyleejordan — APPROVED** [review]
Makes sense to me!

**Comment by daxian-dbw** [comment]
Thanks @andschwa!

**Comment by adityapatwardhan** [comment]
LGTM

**Pull request merged**